lib/stores/file_tree.js

```diff
@@ -157,39 +159,68 @@ class FileTree {
     return versions.find(version => modified === version.trim().replace(/"/g, ''));
   }
 
+  async _etagMatchOrThrow (originalEtag, username, pathname, isdir) {
+    const metadata = await this.readMeta(username, pathname, isdir);
+    if (metadata.ETag !== originalEtag) {
+      throw new Error('ETag mismatch');
+    }
+  }
+
+  async _delay (t) {
+    return new Promise(resolve => setTimeout(resolve.bind(null), t));
+  }
+
   async get (username, pathname, versions, head = false) {
+    const startTime = parseInt(Date.now());
```

Comment:
Date.now() returns "the number of milliseconds elapsed since the epoch" so `parseInt` is unnecessary here and line 181